src/components/WeekWeatherData/index.js

```diff
@@ -33,6 +34,8 @@ const langMap = {
 export function WeekWeatherData() {
   const { i18n } = useTranslation();
   const dispatch = useDispatch();
+  const localStoreLanguage =
+    JSON.parse(localStorage.getItem('language')) || undefined;
```

Comment:
This can also be moved to a helper function, as it's the same in 2 places.
Also, it would be a good idea to have all functions accessing `localStorage` in one file.